src/item.cpp

```diff
@@ -760,15 +760,16 @@ std::string item::info( bool showtext, std::vector<iteminfo> &info ) const
 
         std::string vits;
         for( const auto &v : g->u.vitamins_from( *food_item ) ) {
-            if( v.second != 0 ) {
+            // only display vitamins that we actually require
+            if( g->u.vitamin_rate( v.first ) > 0 && v.second != 0 ) {
                 if( !vits.empty() ) {
                     vits += ", ";
                 }
-                vits += string_format( "%s (%i)", v.first.obj().name().c_str(), v.second );
+                vits += string_format( "%s (%i%%)", v.first.obj().name().c_str(), int( v.second / ( 1440.0 / g->u.vitamin_rate( v.first ) ) * 100 ) );
```

Comment:
`1440 = 60 * 24`

Seems to be the number of some time units it will keep you nourished with that vitamin at current rate.

Should be all expressed in `MINUTES(x)` and `HOURS(x)` macros. Then commented.

Also, the choice of unit (minutes) for `vitamin_rate` is pretty bad. Easy to confuse with turns (`MINUTES(1) == 10`). Changing it to turns would simplify things and prevent bugs.